lib/shopping_cart.rb

```diff
@@ -0,0 +1,40 @@
+class ShoppingCart
+
+  attr_reader :name, :capacity, :products
+  def initialize(name, capacity)
+    @name = name
+    @capacity = capacity
+    @products = []
+  end
+
+  def add_product(product)
+    @products << product
+  end
+
+  def details
+    store_details = {}
+    store_details[:name] = name
+    store_details[:capacity] = capacity.delete("items").to_i
+    store_details
```

Comment:
you could also just have something like: 
```
{name: @name, capacity: capacity.delete("items").to_i}
```

Then you condense this method to one line.

Reply:
Also, if you're inside a class, use the instance variable for @ capacity instead of the method that came from the attr_reader.. that allows you to skip a step in the lookup.